File: src/prod/src/ServiceModel/management/FaultAnalysisService/CancelTestCommandDescription.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------


#include "stdafx.h"

using namespace Common;
using namespace Naming;
using namespace ServiceModel;
using namespace std;

using namespace Management::FaultAnalysisService;

StringLiteral const TraceComponent("CancelTestCommandDescription");

CancelTestCommandDescription::CancelTestCommandDescription() 
    : operationId_()
    , force_(false)
{ 
}

CancelTestCommandDescription::CancelTestCommandDescription(Guid &operationId, bool force)
: operationId_(operationId)
, force_(force)
{
}


CancelTestCommandDescription::CancelTestCommandDescription(CancelTestCommandDescription const & other)
: operationId_(other.Id)
, force_(other.Force)
{
}

CancelTestCommandDescription::CancelTestCommandDescription(CancelTestCommandDescription && other) 
    : operationId_(move(other.operationId_))
    , force_(move(other.force_))
{ 
}

bool CancelTestCommandDescription::operator == (CancelTestCommandDescription const & other) const
{
    if (operationId_ != other.operationId_) { return false; }

    if (force_ != other.force_) { return false; }

    return true;
}

bool CancelTestCommandDescription::operator != (CancelTestCommandDescription const & other) const
{
    return !(*this == other);
}

void CancelTestCommandDescription::WriteTo(TextWriter & w, FormatOptions const &) const
{
   w << "[ " << operationId_;  
   
   w << ", " << force_ << " ]";
}

Common::ErrorCode CancelTestCommandDescription::FromPublicApi(
    FABRIC_CANCEL_TEST_COMMAND_DESCRIPTION const & publicDescription)
{
    operationId_ = Guid(publicDescription.OperationId);

    force_ = publicDescription.Force ? true : false;
    
    return ErrorCodeValue::Success;
}

File: src/prod/src/managed/Api/src/System/Fabric/Description/CancelTestCommandDescription.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace System.Fabric.Description
{
    using System;
    using System.Fabric.Common;
    using System.Fabric.Interop;
    using System.Fabric.Strings;
    using System.Globalization;

    internal sealed class CancelTestCommandDescription
    {
        public CancelTestCommandDescription(
            Guid operationId,
            bool force)
        {
            Requires.Argument<Guid>("operationId", operationId).NotNull();
            this.OperationId = operationId;
            this.Force = force;
        }

        public Guid OperationId
        {
            get;
            internal set;
        }

        public bool Force
        {
            get;
            internal set;
        }

        public override string ToString()
        {
            return string.Format(CultureInfo.InvariantCulture, "OperationId: {0}, Force: {1}",
                this.OperationId,
                this.Force);
        }

        internal IntPtr ToNative(PinCollection pinCollection)
        {
            var nativeDescription = new NativeTypes.FABRIC_CANCEL_TEST_COMMAND_DESCRIPTION();

            nativeDescription.OperationId = this.OperationId;
            nativeDescription.Force = NativeTypes.ToBOOLEAN(this.Force);

            return pinCollection.AddBlittable(nativeDescription);
        }


        internal static unsafe CancelTestCommandDescription CreateFromNative(IntPtr nativeRaw)
        {
            NativeTypes.FABRIC_CANCEL_TEST_COMMAND_DESCRIPTION native = *(NativeTypes.FABRIC_CANCEL_TEST_COMMAND_DESCRIPTION*)nativeRaw;

            Guid operationId = native.OperationId;            

            bool force = NativeTypes.FromBOOLEAN(native.Force);

            return new CancelTestCommandDescription(operationId, force);
        }

    }
}
